src/Controller/Api/Legacy/GroupController.php

```diff
@@ -0,0 +1,256 @@
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * @author Hugo Briand <briand@ekino.com>
+ */
+class GroupController
+{
+    /**
+     * @var GroupManagerInterface
+     */
+    protected $groupManager;
+
+    /**
+     * @var FormFactoryInterface
+     */
+    protected $formFactory;
+
+    /**
+     * @param GroupManagerInterface $groupManager Sonata group manager
+     * @param FormFactoryInterface  $formFactory  Symfony form factory
+     */
+    public function __construct(GroupManagerInterface $groupManager, FormFactoryInterface $formFactory)
+    {
+        $this->groupManager = $groupManager;
+        $this->formFactory = $formFactory;
+    }
+
+    /**
+     * Returns a paginated list of groups.
+     *
+     * @ApiDoc(
+     *  resource=true,
+     *  output={"class"="Sonata\DatagridBundle\Pager\PagerInterface", "groups"={"sonata_api_read"}}
+     * )
+     *
+     * @QueryParam(name="page", requirements="\d+", default="1", description="Page for groups list pagination (1-indexed)")
+     * @QueryParam(name="count", requirements="\d+", default="10", description="Number of groups per page")
+     * @QueryParam(name="orderBy", map=true, requirements="ASC|DESC", nullable=true, strict=true, description="Query groups order by clause (key is field, value is direction")
+     * @QueryParam(name="enabled", requirements="0|1", nullable=true, strict=true, description="Enables or disables the groups only?")
+     *
+     * @View(serializerGroups={"sonata_api_read"}, serializerEnableMaxDepthChecks=true)
+     *
+     * @return PagerInterface
+     */
+    public function getGroupsAction(ParamFetcherInterface $paramFetcher)
+    {
+        $supportedFilters = [
+            'enabled' => '',
+        ];
+
+        $page = $paramFetcher->get('page');
+        $limit = $paramFetcher->get('count');
+        $sort = $paramFetcher->get('orderBy');
+        $criteria = array_intersect_key($paramFetcher->all(), $supportedFilters);
+
+        $criteria = array_filter($criteria, static function ($value): bool {
+            return null !== $value;
+        });
+
+        if (!$sort) {
+            $sort = [];
+        } elseif (!\is_array($sort)) {
+            $sort = [$sort, 'asc'];
+        }
+
+        return $this->groupManager->getPager($criteria, $page, $limit, $sort);
+    }
+
+    /**
+     * Retrieves a specific group.
+     *
+     * @ApiDoc(
+     *  requirements={
+     *      {"name"="id", "dataType"="string", "description"="Group identifier"}
+     *  },
+     *  output={"class"="FOS\UserBundle\Model\GroupInterface", "groups"={"sonata_api_read"}},
+     *  statusCodes={
+     *      200="Returned when successful",
+     *      404="Returned when group is not found"
+     *  }
+     * )
+     *
+     * @View(serializerGroups={"sonata_api_read"}, serializerEnableMaxDepthChecks=true)
+     *
+     * @param string $id
+     *
+     * @return GroupInterface
+     */
```

Comment:
Can't we use type declarations here?

Reply:
We can but we shouldn't. To keep BC-break as minimal as it is possible. This PR move `Controller\Api` to `Controller\Api\Legacy`, so it is not new file.

BTW This legacy code will be remove in next major.

Reply:
Ok if it gets remove, let's not bother you with that.